src/_posts/2019-02-07-working-with-dynamodb.md

```diff
@@ -0,0 +1,995 @@
+
+            builder.attributeDefinitions(
+                AttributeDefinition.builder()
+                    .attributeName("task_id")
+                    .attributeType(ScalarAttributeType.N)
+                    .build()
+            )
+        }
+    }
+}
+```
+
+So, what's going on in this `createTable` method? Let's break down command by command and see:
+
+```kotlin
+builder.tableName("tasqui")
+```
+
+This is a fairly easy part, we are just setting the name of the table, then we have:
+
+```kotlin
+builder.provisionedThroughput { provisionedThroughput ->
+    provisionedThroughput.readCapacityUnits(5)
+    provisionedThroughput.writeCapacityUnits(5)
+}
+```
+
+This part is seeting the throughput for the table, which is the ability to read and write things to the db. We are setting the read and write throughput to 5, but what 5 means exactly? How the throughput is calculated? 
+
+The throughput is measured in `units`, each `unit` might have different values depending in which kind of operation you are doing. For reads, each `unit` is 4Kb/s for consistently strong read, and 8Kb/s for eventually consistent. While writing things get a bit easier, 1 `unit` is 1Kb/s and you don't have any difference between strong or eventual consistency. 
+
+In this case, 5 was chosen since is the default value that Amazon gives to you in the free tier.  
+
+Moving to our actual table, we have to set the Primary Key:
+
+```kotlin
+builder.keySchema(
+    KeySchemaElement.builder()
+        .attributeName("task_id")
+        .keyType(KeyType.HASH)
+        .build()
+)
+
+builder.attributeDefinitions(
+    AttributeDefinition.builder()
+        .attributeName("task_id")
+        .attributeType(ScalarAttributeType.N)
+        .build()
+)
+```
+
+This sets the primary key to be named `task_id` and to have a `Partition Key` only by defining the `keyType` to `HASH`, then we set the type of our key, in this case, is an `integer` so we set as `ScalarAttributeType.N`. You can also set has a `string` or `binary`.
+
+Now everything is ready we can start moving to write our assertion. We want the repository to save a task in the database, so we can query for the object that we just saved to see if he is really there. 
+
+```kotlin
+
+class DynamoDbTaskRepositoryShould {
+
+    @Test
+    internal fun `add Task to DynamoDB`() {
+        ...
+        val task = Task(1, "Task description")
+
+        val item = dynamoDbClient.getItem(
+                GetItemRequest.builder()
+                    .tableName("tasqui")
+                    .key(mapOf("task_id" to AttributeValue.builder().n("1").build()))
+                    .build()).item()
+
+        val storedTask = Task(item["task_id"]!!.n().toInt(), item["description"]!!.s())
+
+        Assertions.assertEquals(storedTask, task)
+    }
+}
+```
+
+The sdk provides the method `getItem` to query items specific records from the database, we have to build a `GetItemRequest` passing the `tableName` and the `key`. 
```

Comment:
>The sdk provides the method `getItem` to query items specific records from the database,

The sdk provides the method `getItem` to query specific records from the database,